### PR TITLE
Improve release process docs

### LINKED ISSUE #437: `appVersion` in Chart.yaml should be updated with each release
`appVersion` in Chart.yaml should be updated with each release

> Shouldn't the Chart.yaml be updated with each release?

_Originally posted by @wccropper in https://github.com/trinodb/trino-gateway/issues/435#issuecomment-2289091742_

Ref:
https://github.com/trinodb/trino-gateway/blob/112c500321805de962f6d9874ec5145c70dc8631/helm/trino-gateway/Chart.yaml#L24
https://github.com/trinodb/trino-gateway/blob/112c500321805de962f6d9874ec5145c70dc8631/helm/trino-gateway/values.yaml#L6-L7

### PR DESCRIPTION
**Pull request by mosabua**
## Description

as discussed in dev sync

Fixes #437

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -65,9 +65,18 @@ notes and other details are available on GitHub](https://github.com/trinodb/trin
...
 ## Release process
 
-A full release process consists of the following steps:
+A full release process consists a number of steps:
 
-Merge the pull request for the release notes and pull the changes locally:
+Create a release notes pull request including the following changes:
+
+- Add new release notes in `docs/release-notes.md`.
+- Update version in `docs/quickstart.md`.
```
Comment (oneonestar) [inline docs/development.md:73]:
```suggestion
- Update `VERSION` in `docs/quickstart.md`.
```

File: docs/development.md [hunk 2]
```diff
@@ -65,9 +65,18 @@ notes and other details are available on GitHub](https://github.com/trinodb/trin
...
+- Update `appVersion` to new version and `version` to new version with `.0.0`
+  appended in `helm/trino-gateway/Chart.yaml`
```
Comment (oneonestar) [inline docs/development.md:75]: Maybe? not sure.
```suggestion
- Update `appVersion` to new version and `version` to `<version>.0.0`
  in `helm/trino-gateway/Chart.yaml`
```

Comment (mosabua) [reply docs/development.md:75]: No .. I think thats less explicit. Added an example.

**Comment by mosabua** [comment]
Should be ready again @oneonestar .. also cc @martint

**Review by oneonestar — APPROVED** [review]

**Pull request merged**